lib/kennel.rb

```diff
@@ -169,11 +179,13 @@ def filter_resources!(resources, by, against, name, env)
     end
 
     def load_all
-      loader = Zeitwerk::Loader.new
-      Dir.exist?("teams") && loader.push_dir("teams", namespace: Teams)
-      Dir.exist?("parts") && loader.push_dir("parts")
-      loader.setup
-      loader.eager_load # TODO: this should not be needed but we see hanging CI processes when it's not added
+      @loaded_zeitwerk ||= begin
```

Comment:
add a comment explaining why we need to cache zeitwerk but re-run the require

Reply:
Because it doesn't work otherwise?

I know that's not a very helpful answer. I don't really understand, or trust, Zeitwerk.

Reply:
Something like "we load this multiple times when doing X"
also putting the require into the block, or doing a
```
return if @loaded
@loaded = true
```
would not re-run the require and make it a bit cleaner ?

Reply:
IIRC the error was something to do with there already being a loader registered. 🤷‍♀️

Reply:
yeah, so the method gets called twice, we should find out why and either stop it or make the whole method re-executable (the require still costs time and is wasteful to execute twice)

Reply:
Sure. That's not something that gets _worse_ with this PR though, right? This PR simply makes the problem more _visible_.

Reply:
I'd say this PR makes the problem less visible since it allows ignoring it.
Before the invalid usage would fail.
I'd prefer to either avoid the invalid usage or make it not invalid anymore.
(as in: don't call this code twice or make the whole code hide behind `return if @loaded`, but not making half the code not execute)